lib/kafka/consumer.rb

```diff
@@ -126,7 +130,15 @@ def resume(topic, partition)
     # @param partition [Integer]
     # @return [Boolean] true if the partition is paused, false otherwise.
     def paused?(topic, partition)
-      @paused_partitions.fetch(topic, []).include?(partition)
+      partitions = @paused_partitions.fetch(topic, {})
+
+      if partitions.key?(partition)
+        # Users can set an optional timeout, after which the partition is
+        # automatically resumed. When pausing, the timeout is translated to an
+        # absolute point in time.
+        timeout = partitions.fetch(partition)
+        timeout.nil? || Time.now < timeout
```

Comment:
How about deleting the entry in the `paused_partitions`-hash if the timeout has passed?